### PR TITLE
Improved the readability of code

### PR DESCRIPTION
**Pull request by Avisheet**
The code enhancements include comments for parameter declarations, pipeline setup, and the thread function for processing OakD spatial detections. The comments provide clarity on parameter purposes, node creation, and debugging visualization using OpenCV. Additionally, proper thread termination in the destructor is ensured, and the ROS 2 component registration process is explained for improved code comprehension.

### CODE REVIEW

File: rmf_human_detector_oakd/src/HumanDetector.cpp [hunk 1]
```diff
@@ -98,7 +99,7 @@ HumanDetector::HumanDetector(
   xoutBoundingBoxDepthMapping->setStreamName("boundingBoxDepthMapping");
   xoutDepth->setStreamName("depth");
 
-  // Set properties
+  // Set properties for camera node
```
Comment (Yadunund) [inline rmf_human_detector_oakd/src/HumanDetector.cpp:102]:
```suggestion
  // Set properties for oak-d camera node
```

File: rmf_human_detector_oakd/src/HumanDetector.cpp [hunk 2]
```diff
@@ -112,7 +113,7 @@ HumanDetector::HumanDetector(
     dai::MonoCameraProperties::SensorResolution::THE_720_P);
   monoRight->setBoardSocket(dai::CameraBoardSocket::RIGHT);
 
-  // Setting node configs
+  // Setting stereo depth node configs
```
Comment (Yadunund) [inline rmf_human_detector_oakd/src/HumanDetector.cpp:116]:
```suggestion
  // Setting properties for oak-d stereo depth node
```

File: rmf_human_detector_oakd/src/HumanDetector.cpp [hunk 3]
```diff
@@ -146,6 +147,7 @@ HumanDetector::HumanDetector(
   stereo->depth.link(spatialDetectionNetwork->inputDepth);
   spatialDetectionNetwork->passthroughDepth.link(xoutDepth->input);
 
+// Function for the detection of the thread
```
Comment (Yadunund) [inline rmf_human_detector_oakd/src/HumanDetector.cpp:150]: Not sure what you mean by "detection of the thread"? Did you mean
```suggestion
// Run detection in a separate thread
```

File: rmf_human_detector_oakd/src/HumanDetector.cpp [hunk 4]
```diff
@@ -28,6 +28,7 @@
 namespace rmf_human_detector_oakd {
 
 //==============================================================================
+// constructor called
```
Comment (Yadunund) [inline rmf_human_detector_oakd/src/HumanDetector.cpp:31]: I don't think `constructor called` is the right comment here. It's merely the definition of the constructor.
```suggestion
// Constructor
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for the PR. Left some feedback. 

> Additionally, proper thread termination in the destructor is ensured, and the ROS 2 component registration process is explained for improved code comprehension.

I don't see any changes relevant to this statement in the PR description. Is this PR still a work in progress?

**Comment by Avisheet** [comment]
Thanks @Yadunund for suggesting the changes. The work is in still progress I'm trying to make the comments so that it will more readable . Further it will help me to understand this project and will be able to contribute in this project

**Comment by Avisheet** [comment]
I was getting DCO error continuously so i am closing this PR. I will create a new branch to work with.

**Pull request closed without merging**